<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from robot.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="robot_ia">
  <!-- Specify some standard inertial calculations https://en.wikipedia.org/wiki/List_of_moments_of_inertia -->
  <!-- These make use of xacro's mathematical functionality -->
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="orange">
    <color rgba="1 0.3 0.1 1"/>
  </material>
  <material name="blue">
    <color rgba="0.2 0.2 1 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="red">
    <color rgba="1 0 0 1"/>
  </material>
  <material name="grey">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <!-- BASE LINK -->
  <link name="base_link">
  </link>
  <!-- BASE_FOOTPRINT LINK -->
  <joint name="base_footprint_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base_footprint"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="base_footprint">
    </link>

  <joint name="lidar_joint" type="fixed">
    <parent link="base_link"/>
    <child link="lidar_frame"/>
    <origin xyz="0 0 0"/>
  </joint>

    <link name="lidar_frame">
  </link>
  <!-- CHASSIS LINK -->
  <joint name="chassis_joint" type="fixed">
    <parent link="base_link"/>
    <child link="chassis"/>
    <origin xyz="0 0 0"/>
  </joint>
  <link name="chassis">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="50"/>
      <inertia ixx="1.2083333333333333" ixy="0.0" ixz="0.0" iyy="1.2083333333333333" iyz="0.0" izz="2.083333333333333"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/ia/works/wheeltec_gazebo/install/ia_robot_urdf/share/ia_robot_urdf/meshes/base_link.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.741176470588235 0.901960784313726 0.956862745098039 1"/>
      </material>
    </visual>
  </link>
  <!-- LEFT WHEEL LINK -->
  <link name="lf_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.02"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.000456492835033346"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/ia/works/wheeltec_gazebo/install/ia_robot_urdf/share/ia_robot_urdf/meshes/lf_wheel.stl"/>
      </geometry>
      <material name="">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <joint name="lf_wheel_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.5 0.25227 0.09"/>
    <parent link="base_link"/>
    <child link="lf_wheel_link"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="lb_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.02"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.000456492835033346"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/ia/works/wheeltec_gazebo/install/ia_robot_urdf/share/ia_robot_urdf/meshes/lb_wheel.stl"/>
      </geometry>
      <material name="">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <joint name="lb_wheel_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.5 0.25227 0.09"/>
    <parent link="base_link"/>
    <child link="lb_wheel_link"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="rb_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.02"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.000456492835033346"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/ia/works/wheeltec_gazebo/install/ia_robot_urdf/share/ia_robot_urdf/meshes/rb_wheel.stl"/>
      </geometry>
      <material name="">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <joint name="rb_wheel_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.5 -0.25227 0.09"/>
    <parent link="base_link"/>
    <child link="rb_wheel_link"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="rf_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.02"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.000456492835033346"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/ia/works/wheeltec_gazebo/install/ia_robot_urdf/share/ia_robot_urdf/meshes/rf_wheel.stl"/>
      </geometry>
      <material name="">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <joint name="rf_wheel_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.5 -0.25227 0.09"/>
    <parent link="base_link"/>
    <child link="rf_wheel_link"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- Left Middle Wheel -->
  <link name="lm_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.02"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.000456492835033346"/>
    </inertial>
    <collision>
      <origin rpy="-1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <joint name="lm_wheel_joint" type="continuous">
    <origin rpy="0 0 0" xyz="0.0 0.25227 0.09"/>
    <parent link="base_link"/>
    <child link="lm_wheel_link"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- Right Middle Wheel -->
  <link name="rm_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.02"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.000456492835033346"/>
    </inertial>
    <collision>
      <origin rpy="-1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <joint name="rm_wheel_joint" type="continuous">
    <origin rpy="0 0 0" xyz="0.0 -0.25227 0.09"/>
    <parent link="base_link"/>
    <child link="rm_wheel_link"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- <xacro:include filename="gazebo_control.xacro" /> -->
  <joint name="camera_joint" type="fixed">
    <parent link="chassis"/>
    <child link="camera_link"/>
    <origin rpy="0 0.5235987755982988 0" xyz="0.05 0 1.18"/>
  </joint>
  <link name="camera_link">
    <visual>
      <geometry>
        <box size="0.0010 0.01 0.01"/>
      </geometry>
      <material name="red"/>
    </visual>
  </link>
  <joint name="camera_optical_joint" type="fixed">
    <parent link="camera_link"/>
    <child link="camera_link_optical"/>
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
  </joint>
  <link name="camera_link_optical"/>
</robot>
